WalletWasabi/Crypto/Strobe.cs

```diff
@@ -0,0 +1,318 @@
+using System;
+using System.Runtime.InteropServices;
+using System.Text;
+
+namespace WalletWasabi.Crypto
+{
+	// https://strobe.sourceforge.io/papers/strobe-20170130.pdf
+	// Based on Merlin framework small implementation: https://doc-internal.dalek.rs/src/merlin/strobe.rs.html
+	public class Strobe128
+	{
+		private readonly static byte DDATA = 0x04; // for compatibility with cSHAKE (?)
+		private readonly static byte DRATE = 0x80; // for compatibility with cSHAKE (?)
+
+		private readonly byte[] _state = new byte[200];
+		private byte _position = 0;
+		private byte _beginPosition = 0;
+		private StrobeFlags _currentFlags = 0;
+		private readonly static byte SpongeRate = 166; //Let ˆr=r/8−2.  This is the portion of the rate which is used for user data,measured in bytes.
+		// private readonly static byte SpongeCapacity = 0;
+
+		public Strobe128(string procotol)
+		{
+			var V = "STROBEv1.0.2";
+			var S0 = ByteHelpers.Combine(
+				new byte[]{ 1, (byte)(SpongeRate + 2), 1, 0, 1, 12 * 8 },  // F([[1, r/8, 1, 0, 1, 12·8]]
+				Encoding.UTF8.GetBytes(V));
+			Buffer.BlockCopy(S0, 0, _state, 0, S0.Length);
+			KeccakF1600(_state);
+			MetaAssociatedData(Encoding.UTF8.GetBytes(procotol), false);
+		}
+
+		public void MetaAssociatedData(byte[] data, bool more)
+		{
+			BeginOperation(StrobeFlags.M | StrobeFlags.A, more);
+			Absorb(data);
+		}
+
+		public void AssociatedData(byte[] data, bool more)
```

Comment:
not to rehash the same argument, but I'm actually in favor of `AddAssociatedData`, the point about method names involving a verb was well taken

Reply:
Merlin uses `ad` while the specs name this `AssociatedData`. Both are really bad names to me. I replaced it by `AddData` and `AddMetaData` when the flag is `M`. These new names are clearer but now we have ugly names as `Prf` and `Key`. Honestly I don't want to diverge too much from the specs and the rust code taken as reference so, I would let it as it is.

Reply:
i feel very strongly about making the noun that is added `AssociatedData` or `AD`, as that is a standard term in cryptography (even though it's kind of a crap term, but it comes form "authenticated encryption with associated data", aka AEAD modes of encryption)

Reply:
FWIW I think `MixKey` (Noise's name for this operation) or `AddKey` and `GeneratePrf` or `OutputPrf` would be just fine if we want verbs instead of declarative operation names, so long as the method names only add to the corresponding STROBE operations, but don't change the operation name itself.